pyuvdata/uvbeam.py

```diff
@@ -0,0 +1,1023 @@
+                                           description=desc,
+                                           form=('Nelements',),
+                                           expected_type=np.float)
+
+        desc = ('Required if antenna_type = "phased_array". Matrix of complex '
+                'element couplings, units: dB, '
+                'shape: (Nelements, Nelements, Nfeed, Nfeed, Nspws, Nfreqs)')
+        self._coupling_matrix = uvp.UVParameter('coupling_matrix', required=False,
+                                                description=desc,
+                                                form=('Nelements', 'Nelements',
+                                                      'Nfeed', 'Nfeed', 'Nspws', 'Nfreqs'),
+                                                expected_type=np.complex)
+
+        # -------- extra, non-required parameters ----------
+        desc = ('Any user supplied extra keywords, type=dict')
+        self._extra_keywords = uvp.UVParameter('extra_keywords', required=False,
+                                               description=desc, value={},
+                                               spoof_val={}, expected_type=dict)
+
+        desc = 'Array of system temperatures, shape (Nspws, Nfreqs), units K'
+        self._system_temperature_array = \
+            uvp.UVParameter('system_temperature_array', required=False,
+                            description=desc, form=('Nspws', 'Nfreqs'),
+                            expected_type=np.float, tols=1e-3)
+
+        desc = 'Array of antenna losses, shape (Nspws, Nfreqs), units dB?'
+        self._loss_array = uvp.UVParameter('loss_array', required=False,
+                                           description=desc, form=('Nspws', 'Nfreqs'),
+                                           expected_type=np.float,
+                                           tols=1e-3)
+
+        desc = 'Array of antenna-amplifier mismatches, shape (Nspws, Nfreqs), units ?'
+        self._mismatch_array = uvp.UVParameter('mismatch_array', required=False,
+                                               description=desc,
+                                               form=('Nspws', 'Nfreqs'),
+                                               expected_type=np.float,
+                                               tols=1e-3)
+
+        desc = ('S parameters of receiving chain, shape (4, Nspws, Nfreqs), '
+                'ordering: s11, s12, s21, s22. units ?')
+        self._s_parameters = uvp.UVParameter('s_parameters', required=False,
+                                             description=desc,
+                                             form=(4, 'Nspws', 'Nfreqs'),
+                                             expected_type=np.float,
+                                             tols=1e-3)
+
+        super(UVBeam, self).__init__()
+
+    def check(self, run_check_acceptability=True):
+        """
+        Add some extra checks on top of checks on UVBase class.
```

Comment:
Are there meant to be more checks other than set_cs_params? If not, we can drop this line of the doc string.